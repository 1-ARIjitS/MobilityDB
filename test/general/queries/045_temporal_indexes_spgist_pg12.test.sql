-------------------------------------------------------------------------------
--
-- This MobilityDB code is provided under The PostgreSQL License.
-- Copyright (c) 2016-2022, Université libre de Bruxelles and MobilityDB
-- contributors
--
-- MobilityDB includes portions of PostGIS version 3 source code released
-- under the GNU General Public License (GPLv2 or later).
-- Copyright (c) 2001-2022, PostGIS contributors
--
-- Permission to use, copy, modify, and distribute this software and its
-- documentation for any purpose, without fee, and without a written
-- agreement is hereby granted, provided that the above copyright notice and
-- this paragraph and the following two paragraphs appear in all copies.
--
-- IN NO EVENT SHALL UNIVERSITE LIBRE DE BRUXELLES BE LIABLE TO ANY PARTY FOR
-- DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING
-- LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION,
-- EVEN IF UNIVERSITE LIBRE DE BRUXELLES HAS BEEN ADVISED OF THE POSSIBILITY
-- OF SUCH DAMAGE.
--
-- UNIVERSITE LIBRE DE BRUXELLES SPECIFICALLY DISCLAIMS ANY WARRANTIES,
-- INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
-- AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS ON
-- AN "AS IS" BASIS, AND UNIVERSITE LIBRE DE BRUXELLES HAS NO OBLIGATIONS TO
-- PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS. 
--
-------------------------------------------------------------------------------

ANALYZE tbl_tint;
DROP INDEX IF EXISTS tbl_tint_spgist_idx;
CREATE INDEX tbl_tint_spgist_idx ON tbl_tint USING SPGIST(temp);

SELECT k, temp |=| intrange '[100,100]'::tbox FROM tbl_tint ORDER BY 2, 1 LIMIT 3;
SELECT k, temp |=| tint '[1@2001-06-01, 2@2001-07-01]' FROM tbl_tint ORDER BY 2, 1 LIMIT 3;

DROP INDEX tbl_tint_spgist_idx;

-------------------------------------------------------------------------------

ANALYZE tbl_tfloat;
DROP INDEX IF EXISTS tbl_tfloat_spgist_idx;
CREATE INDEX tbl_tfloat_spgist_idx ON tbl_tfloat USING SPGIST(temp);

SELECT k, round((temp |=| floatrange '[100,100]'::tbox)::numeric, 6) FROM tbl_tfloat ORDER BY 2, 1 LIMIT 3;
SELECT k, round((temp |=| tfloat '[1.5@2001-06-01, 2.5@2001-07-01]')::numeric, 6) FROM tbl_tfloat ORDER BY 2, 1 LIMIT 3;

DROP INDEX tbl_tfloat_spgist_idx;

-------------------------------------------------------------------------------
