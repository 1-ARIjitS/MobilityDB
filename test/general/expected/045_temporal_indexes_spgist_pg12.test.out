ANALYZE tbl_tint;
ANALYZE
DROP INDEX IF EXISTS tbl_tint_spgist_idx;
NOTICE:  index "tbl_tint_spgist_idx" does not exist, skipping
DROP INDEX
CREATE INDEX tbl_tint_spgist_idx ON tbl_tint USING SPGIST(temp);
CREATE INDEX
SELECT k, temp |=| intrange '[100,100]'::tbox FROM tbl_tint ORDER BY 2, 1 LIMIT 3;
 k  | ?column? 
----+----------
 55 |        1
 64 |        1
 82 |        1
(3 rows)

SELECT k, temp |=| tint '[1@2001-06-01, 2@2001-07-01]' FROM tbl_tint ORDER BY 2, 1 LIMIT 3;
 k  | ?column? 
----+----------
 84 |        2
 85 |        2
 27 |        5
(3 rows)

DROP INDEX tbl_tint_spgist_idx;
DROP INDEX
ANALYZE tbl_tfloat;
ANALYZE
DROP INDEX IF EXISTS tbl_tfloat_spgist_idx;
NOTICE:  index "tbl_tfloat_spgist_idx" does not exist, skipping
DROP INDEX
CREATE INDEX tbl_tfloat_spgist_idx ON tbl_tfloat USING SPGIST(temp);
CREATE INDEX
SELECT k, round((temp |=| floatrange '[100,100]'::tbox)::numeric, 6) FROM tbl_tfloat ORDER BY 2, 1 LIMIT 3;
 k  |  round   
----+----------
 72 | 0.082398
 77 | 0.216814
 82 | 0.393097
(3 rows)

SELECT k, round((temp |=| tfloat '[1.5@2001-06-01, 2.5@2001-07-01]')::numeric, 6) FROM tbl_tfloat ORDER BY 2, 1 LIMIT 3;
 k  |  round   
----+----------
 44 | 0.382978
 45 | 2.851887
 86 | 3.794321
(3 rows)

DROP INDEX tbl_tfloat_spgist_idx;
DROP INDEX
